src/states/elder/mod.rs

```diff
@@ -629,14 +629,14 @@ impl Elder {
         let dst = Authority::Node(*pub_id.name());
 
         // Make sure we are connected to the candidate
-        if !self.peer_map().has(&pub_id) {
+        if !self.peer_map().has(p2p_node.peer_addr()) {
             trace!(
                 "{} - Not yet connected to {} - use p2p_node.",
                 self,
                 p2p_node
             );
             self.peer_map_mut()
-                .insert(pub_id, p2p_node.connection_info().clone());
+                .connect(p2p_node.connection_info().clone());
```

Comment:
OK, this might be ok for now.
Ideally I'd think these would be gone (i.e only update the peer map on quic_p2p event or disconnect call), does it fail if we remove the following?
I'm thinking we should get a `ConnectedTo` event from quic_p2p when sending that?

```
             self.peer_map_mut()
                .connect(p2p_node.connection_info().clone()); 
```

Same apply for all `connect` calls before a `send_direct_message`

Reply:
Removing those `connect` calls does still seem to pass the tests